Mantid won't build with SIP 4.19

<!-- TEMPLATE FOR BUG REPORTS -->

### Expected behavior

### Actual behavior

```
[3/351] Building CXX object MantidQt/Python/CMakeFiles/mantidqtpython.dir/sip_mantidqt.cpp.o
FAILED: MantidQt/Python/CMakeFiles/mantidqtpython.dir/sip_mantidqt.cpp.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++   -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG -DENABLE_OPENCASCADE -DHAVE_STDINT_H -DMAKE_VATES -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_SVG_LIB -DQT_XML_LIB -Dmantidqtpython_EXPORTS -isystem /usr/local/include -isystem /usr/local/include/nexus -I/Users/svh/Documents/MantidProject/mantid/Framework/Kernel/inc -I/Users/svh/Documents/MantidProject/mantid/Framework/HistogramData/inc -I/Users/svh/Documents/MantidProject/mantid/Framework/Indexing/inc -I/Users/svh/Documents/MantidProject/mantid/Framework/Geometry/inc -I/Users/svh/Documents/MantidProject/mantid/Framework/API/inc -isystem /usr/local/Cellar/qt/4.8.7_2/include -iframework /usr/local/Cellar/qt/4.8.7_2/lib -isystem /usr/local/Cellar/qt/4.8.7_2/include/QtOpenGL -isystem /usr/local/Cellar/qt/4.8.7_2/include/QtSvg -isystem /usr/local/Cellar/qt/4.8.7_2/include/QtGui -isystem /usr/local/Cellar/qt/4.8.7_2/include/QtXml -isystem /usr/local/Cellar/qt/4.8.7_2/lib/QtCore.framework/Headers -I/Users/svh/Documents/MantidProject/mantid/MantidQt/API/inc -IMantidQt/API -I/Users/svh/Documents/MantidProject/mantid/MantidQt/MantidWidgets/inc -IMantidQt/MantidWidgets -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -I/Users/svh/Documents/MantidProject/mantid/MantidQt/Python -I/Users/svh/Documents/MantidProject/mantid/MantidQt/Python/../SliceViewer/inc -I/Users/svh/Documents/MantidProject/mantid/MantidQt/Python/../MantidWidgets/inc -I/Users/svh/Documents/MantidProject/mantid/MantidQt/Python/../MantidWidgets/inc/MantidQtMantidWidgets -I/Users/svh/Documents/MantidProject/mantid/MantidQt/Python/../MantidWidgets/inc/MantidQtMantidWidgets/InstrumentView -I/Users/svh/Documents/MantidProject/mantid/MantidQt/Python/../Factory/inc -I/Users/svh/Documents/MantidProject/mantid/MantidQt/Python/../SpectrumViewer/inc -I/Users/svh/Documents/MantidProject/mantid/MantidQt/Python/../RefDetectorViewer/inc -IMantidQt/Python/../SliceViewer -IMantidQt/Python/../RefDetectorViewer -Wall -Wextra -Wconversion -Winit-self -Wpointer-arith -Wcast-qual -Wcast-align -fno-common -Wno-deprecated -Wno-write-strings -Wno-sign-conversion -Woverloaded-virtual -fno-operator-names -O2 -g -DNDEBUG -fPIC   -std=gnu++14 -MD -MT MantidQt/Python/CMakeFiles/mantidqtpython.dir/sip_mantidqt.cpp.o -MF MantidQt/Python/CMakeFiles/mantidqtpython.dir/sip_mantidqt.cpp.o.d -o MantidQt/Python/CMakeFiles/mantidqtpython.dir/sip_mantidqt.cpp.o -c /Users/svh/Documents/MantidProject/build/MantidQt/Python/sip_mantidqt.cpp
In file included from /Users/svh/Documents/MantidProject/build/MantidQt/Python/sip_mantidqt.cpp:18:
/Users/svh/Documents/MantidProject/build/MantidQt/Python/sipmantidqtpythonpart0.cpp:16365:27: error: call to implicitly-deleted copy constructor of '::MantidQt::Factory::WidgetFactory'
            sipCpp = new  ::MantidQt::Factory::WidgetFactory(*a0);
                          ^                                  ~~~
/Users/svh/Documents/MantidProject/mantid/MantidQt/Python/../Factory/inc/MantidQtFactory/WidgetFactory.h:48:51: note: copy constructor of 'WidgetFactory' is implicitly deleted because base class 'QObject' has an inaccessible copy constructor
class EXPORT_OPT_MANTIDQT_FACTORY WidgetFactory : public QObject {
                                                  ^
1 error generated.
ninja: build stopped: subcommand failed.
```

### Steps to reproduce the behavior

### Platforms affected

Homebrew already updated to SIP 4.19. Other distributions will likely follow soon.

--- a/MantidPlot/src/PythonScript.cpp
+++ b/MantidPlot/src/PythonScript.cpp
@@ -130,11 +130,8 @@ void PythonScript::setIdentifier(const QString &name) {
  * @return A PyObject wrapping this instance
  */
 PyObject *PythonScript::createSipInstanceFromMe() {
-  static sipWrapperType *sipClass(NULL);
-  if (!sipClass) {
-    sipClass = sipFindClass("PythonScript");
-  }
-  PyObject *sipWrapper = sipConvertFromInstance(this, sipClass, NULL);
+  const sipTypeDef *sipClass = sipFindType("PythonScript");
+  PyObject *sipWrapper = sipConvertFromType(this, sipClass, NULL);
   assert(sipWrapper);
   return sipWrapper;
 }

--- a/MantidPlot/src/PythonScripting.cpp
+++ b/MantidPlot/src/PythonScripting.cpp
@@ -348,10 +348,10 @@ bool PythonScripting::setQObject(QObject *val, const char *name,
   if (!sipAPI__qti->api_find_class) {
     throw std::runtime_error("sipAPI_qti->api_find_class is undefined");
   }
-  sipWrapperType *klass = sipFindClass(val->metaObject()->className());
+  const sipTypeDef *klass = sipFindType(val->metaObject()->className());
   if (!klass)
     return false;
-  pyobj = sipConvertFromInstance(val, klass, NULL);
+  pyobj = sipConvertFromType(val, klass, NULL);
 
   if (!pyobj)
     return false;

--- a/MantidPlot/src/qti.sip
+++ b/MantidPlot/src/qti.sip
@@ -1098,7 +1098,7 @@ class ApplicationWindow: QMainWindow
 // we have to do this to override casting in qt/qobject.sip (PyQt 3.16)
 // Russell Taylor, 4/8/09: Change needed for mac build.
 //   Taken from main qtiplot svn revision 1187.
-sipClass=sipFindClass(sipCpp->metaObject()->className());
+sipType=sipFindType(sipCpp->metaObject()->className());
 %End
 
 public:
@@ -1262,8 +1262,8 @@ ApplicationWindow *sipqti_app()
   PyObject *mydict = PyModule_GetDict(me);
   PyObject *pyapp = PyDict_GetItemString(mydict,"app");
   Py_DECREF(me);
-  if (sipCanConvertToInstance(pyapp, sipClass_ApplicationWindow, SIP_NOT_NONE))
-    return (ApplicationWindow*) sipConvertToInstance(pyapp, sipClass_ApplicationWindow, NULL, SIP_NOT_NONE, NULL, &iserr);
+  if (sipCanConvertToType(pyapp, sipType_ApplicationWindow, SIP_NOT_NONE))
+    return (ApplicationWindow*) sipConvertToType(pyapp, sipType_ApplicationWindow, NULL, SIP_NOT_NONE, NULL, &iserr);
   else
     return NULL;
 }
@@ -1467,7 +1467,7 @@ class MantidUI: QObject
 // we have to do this to override casting in qt/qobject.sip (PyQt 3.16)
 // Russell Taylor, 4/8/09: Change needed for mac build.
 //   Taken from main qtiplot svn revision 1187.
-sipClass=sipFindClass(sipCpp->metaObject()->className());
+sipType=sipFindType(sipCpp->metaObject()->className());
 %End
 
 public:

--- a/MantidQt/Python/mantidqt.sip
+++ b/MantidQt/Python/mantidqt.sip
@@ -1356,7 +1356,7 @@ MantidQt::SliceViewer::SliceViewer* WidgetFactory::createSliceViewer(const QStri
 
 
 private:
-  WidgetFactory(const WidgetFactory&);
+  WidgetFactory(const MantidQt::SliceViewer::WidgetFactory &);
   WidgetFactory();
 %Docstring
 WidgetFactory::WidgetFactory()

--- a/MantidQt/Python/mantidqt.sip
+++ b/MantidQt/Python/mantidqt.sip
@@ -1356,7 +1356,7 @@ MantidQt::SliceViewer::SliceViewer* WidgetFactory::createSliceViewer(const QStri
 
 
 private:
-  WidgetFactory(const MantidQt::SliceViewer::WidgetFactory &);
+  WidgetFactory(const MantidQt::Factory::WidgetFactory&);
   WidgetFactory();
 %Docstring
 WidgetFactory::WidgetFactory()

--- a/MantidQt/Python/mantidqt.sip
+++ b/MantidQt/Python/mantidqt.sip
@@ -1356,6 +1356,7 @@ MantidQt::SliceViewer::SliceViewer* WidgetFactory::createSliceViewer(const QStri
 
 
 private:
+  WidgetFactory(const WidgetFactory&);
   WidgetFactory();
 %Docstring
 WidgetFactory::WidgetFactory()

